src/LandIce/problems/LandIce_StokesFOBase.hpp

```diff
@@ -967,6 +970,48 @@ constructVelocityEvaluators (PHX::FieldManager<PHAL::AlbanyTraits>& fm0,
   ev = Teuchos::rcp(new StokesFOResid<EvalT,PHAL::AlbanyTraits>(*p,dl));
   fm0.template registerEvaluator<EvalT>(ev);
 
+  //--- Shared Parameter for Extreme Event ---//
+
+  p = rcp(new Teuchos::ParameterList("Theta 0"));
+  p->set< Teuchos::RCP<ParamLib> >("Parameter Library", paramLib);
+  param_name = "Theta 0";
+  p->set<std::string>("Parameter Name", param_name);
+  p->set<const Teuchos::ParameterList*>("Parameters List", &params->sublist("Parameters"));
+  p->set<double>("Default Nominal Value", 0.);
+  Teuchos::RCP<PHAL::SharedParameter<EvalT,PHAL::AlbanyTraits,ParamEnum,ParamEnum::Theta_0>> ptr_theta_0;
+  ptr_theta_0 = Teuchos::rcp(new PHAL::SharedParameter<EvalT,PHAL::AlbanyTraits,ParamEnum,ParamEnum::Theta_0>(*p,dl));
+  fm0.template registerEvaluator<EvalT>(ptr_theta_0);
+
+  p = rcp(new Teuchos::ParameterList("Theta 1"));
+  p->set< Teuchos::RCP<ParamLib> >("Parameter Library", paramLib);
+  param_name = "Theta 1";
+  p->set<std::string>("Parameter Name", param_name);
+  p->set<const Teuchos::ParameterList*>("Parameters List", &params->sublist("Parameters"));
+  p->set<double>("Default Nominal Value", 0.);
+  Teuchos::RCP<PHAL::SharedParameter<EvalT,PHAL::AlbanyTraits,ParamEnum,ParamEnum::Theta_1>> ptr_theta_1;
+  ptr_theta_1 = Teuchos::rcp(new PHAL::SharedParameter<EvalT,PHAL::AlbanyTraits,ParamEnum,ParamEnum::Theta_1>(*p,dl));
+  fm0.template registerEvaluator<EvalT>(ptr_theta_1);
+
+  auto rparams = params->sublist("Random Parameters");
+  if (rparams.isParameter("Number Of Parameters")) {
+    int nrparams = rparams.get<int>("Number Of Parameters");
```

Comment:
Minor but I would just do
```
int nrparams = rparams.get("Number Of Parameters",0);
```
and avoid the if statement, saving one level of indentation.

Reply:
Do we want to default to 0, or do we want to fail if "Number Of Parameters" is not defined? I prefer the latter, so I would remove the if check and just do
`int nrparams = rparams.get<int>("Number Of Parameters");`

Reply:
I updated the PR to be as suggested by @mperego .

Reply:
I'm not sure. What you suggest forces to either have a full valid parameter list, or to not have it at all. I don't see a problem having, say, an empty sublist in the yaml file, which can be parsed as "no request for random phys params"... But I'm ok either way.

Reply:
The issue is when you just forget to put the "Number Of Parameters", and you think Albany is including the parameters but in fact it does not.